include/mrdox/Metadata/TemplateArg.hpp

```diff
@@ -0,0 +1,33 @@
+//
+// This is a derivative work. originally part of the LLVM Project.
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2023 Krystian Stasiowski (sdkrystian@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdox
+//
+
+#ifndef MRDOX_METADATA_TEMPLATEARG_HPP
+#define MRDOX_METADATA_TEMPLATEARG_HPP
+
+#include <mrdox/Platform.hpp>
+#include <string>
+
+namespace clang {
+namespace mrdox {
+
+struct TemplateArgInfo
+{
+    std::string Value;
+
+    TemplateArgInfo() = default;
+
+    TemplateArgInfo(std::string&& value);
```

Comment:
well wouldn't this be `std::string_view` because, does the clang API give you ownership of a `std::string`? Or does it just give you a `llvm::StringRef`?

Reply:
I'm currently using using `TemplateArgument::print` to convert to a string (see https://github.com/cppalliance/mrdox/pull/167/files#diff-1f731482616430ef94ef1ca04a1653124fd230aaa408167e9c9913db990a0ae6R399). This delegates to `TypePrinter` (which uses a visitor to create a string representation of the argument).